tests/unit/scene/test_scene_dunder_methods.py

```diff
@@ -0,0 +1,65 @@
+import arcade
+import pytest
+
+
+def test_len():
+    scene = arcade.Scene()
+    scene.add_sprite_list("Player")
+
+    assert len(scene) == 1
+
+    scene.add_sprite_list("Walls")
+    scene.add_sprite_list("Coins")
+
+    assert len(scene) == 3
+
+def test_delitem():
+    scene = arcade.Scene()
+    scene.add_sprite_list("Walls")
+    scene.add_sprite_list("Player")
+    scene.add_sprite_list("Coins")
+    del scene["Player"]
+
+    with pytest.raises(KeyError):
+        scene["Player"]
+
+    del scene[scene._sprite_lists[0]]
+
+    with pytest.raises(KeyError):
+        scene["Walls"]
+
+    del scene[0]
+
+    with pytest.raises(KeyError):
+        scene["Coins"]
+
+def test_bool():
+    scene = arcade.Scene()
+    assert bool(scene) == False
+
+    scene.add_sprite_list("Walls")
+    assert bool(scene) == True
+
+def test_contains():
+    scene = arcade.Scene()
+    "Walls" not in scene
+    None not in scene
+    
+    Walls_SpriteList = arcade.SpriteList()
+    scene.add_sprite_list("Walls", sprite_list = Walls_SpriteList)
```

Comment:
1. You can drop the space around the `=` when passing keyword arguments
2. I'd add `use_spatial_hash=True` here to make sure we have that condition covered
```suggestion
    scene.add_sprite_list("Walls", use_spatial_hash=True, sprite_list=walls_sprite_list)
```